lms/templates/certificates/_accomplishment-banner.html

```diff
@@ -16,15 +17,15 @@
           });
           $(".action-linkedin-profile").click(function() {
               var data = {
-                  user_id: '${accomplishment_user_id}',
+                  user_id: '${accomplishment_user_id | n, js_escaped_string}',
                   course_id: $(this).data('course-id'),
                   enrollment_mode: $(this).data('certificate-mode'),
-                  certificate_id: '${certificate_id_number}',
+                  certificate_id: '${certificate_id_number | n, js_escaped_string}',
                   certificate_url: window.location.href,
                   social_network: 'LinkedIn'
               };
               Logger.log('edx.certificate.shared', data);
-              window.open('${linked_in_url}');
+              window.open('${linked_in_url | n, decode.utf8}');
```

Comment:
Please apply the js_escaped_string filter to this, as we still need to Javascript/HTML escape URLS. See 
http://edx.readthedocs.io/projects/edx-developer-guide/en/latest/conventions/preventing_xss.html#url-context-and-escaping for more details.

Reply:
Initially i did applied it, but it was breaking one of test. Inspecting into the issue and looking into the url shows that js escaping was creating the issue

without escaped url
---------------------------
``` http://www.linkedin.com/profile/add?_ed=0_0dPSPyS070e0HsE9HNz_13_d11_&amp;pfCertificationName=%C3%A9dX+Honor+Code+Certificate+for+refundable+course&amp;pfCertificationUrl=http%3A%2F%2Ftestserver%2Fcertificates%2Ff61a7536c1f64e93a468ced951805041&amp;source=o&amp;trk=unittest-testorg%2Frun1%2Frefundable_course_honor-dashboard ```

default html escape via <%page expression_filter="h"/>
-------------------------------------------------------------------------
``` http://www.linkedin.com/profile/add?_ed=0_0dPSPyS070e0HsE9HNz_13_d11_&pfCertificationName=%C3%A9dX+Honor+Code+Certificate+for+refundable+course&pfCertificationUrl=http%3A%2F%2Ftestserver%2Fcertificates%2Ff61a7536c1f64e93a468ced951805041 ```

javascript escaped via ${ some_var | n, js_escaped_string }
-------------------------------------------------------------------------------
``` http://www.linkedin.com/profile/add?_ed\\u003Dtest\\u0026pfCertificationName\\u003DTest+Site+Honor+Code+Certificate+for+refundable+course\\u0026pfCertificationUrl\\u003Dhttp%3A%2F%2Ftest\\u002Dsite.testserver%2Fcertificates%2F3cc0343f08734bbc918c8dff2f5487d4\\u0026source\\u003Do\\u0026trk\\u003Dunittest\\u002Dtestorg%2Frun1%2Frefundable_course_honor\\u002Ddashboard\ ```

Reply:
@amangano-edx Removed js_escaped_string as it creates wrong linkedin url, which would not work. Now only html escaping.
```
>>> from django.utils.html import escapejs
>>> escapejs('http://something.com/?a=123&b=adeel+khan')
u'http://something.com/?a\\u003D123\\u0026b\\u003Dadeel+khan'
```